backend/server.go

```diff
@@ -60,34 +62,40 @@ func WrapApiEndpoint(
 	return func(responseWriter http.ResponseWriter, request *http.Request) {
 		allowCors(responseWriter, request)
 
-		var httpStatus int
-		if err, httpStatus := apiHandlerFunc(responseWriter, request); err != nil {
-			if httpStatus >= 500 {
+		body, status, err := apiHandlerFunc(responseWriter, request)
+		if err != nil {
+			responseWriter.WriteHeader(status)
+			if status >= 500 {
 				log.Print(err)
 			}
-			json.NewEncoder(responseWriter).Encode(map[string]string{
-				"error": err.Error(),
-			})
-			responseWriter.WriteHeader(httpStatus)
+
+			if body == nil {
```

Comment:
This line, line 67 and line 77 can all be outside of if err!= nil

Reply:
changed it to this
```
	return func(responseWriter http.ResponseWriter, request *http.Request) {
		allowCors(responseWriter, request)

		body, status, err := apiHandlerFunc(responseWriter, request)
		responseWriter.WriteHeader(status)
		if err != nil {
			if status >= 500 {
				log.Print(err)
			}

			if body == nil {
				body = map[string]interface{}{
					"error": err.Error(),
				}
			}

		} else if body == nil {
			body = emptyJson
		}

		json.NewEncoder(responseWriter).Encode(body)
	}
```

* moved line 67 outside
* this line (72) kept inside because this is all happening in an `if err != nil` clause. We only want to return  `{"error": ...}` to the frontend if there's an error. I added an `else if` to handle nil bodies in the successful case

Reply:
I was thinking of having if body == nil create new JSON and not nested if error != nil assign body["error"]: err.Error(). Is there some advantage of forcing when an error happens that it wipes the body?